tests/test_models.py

```
@@ -0,0 +1,755 @@
+"""
+Tests for Tidal Data Models
+
+Comprehensive unit tests for data models including Artist, Album, Track, Playlist,
+and SearchResults. Tests cover initialization, serialization, validation, and edge cases.
+"""
+
+import pytest
+from dataclasses import FrozenInstanceError
```

Ruff (0.12.2)

`dataclasses.FrozenInstanceError` imported but unused

Remove unused import: `dataclasses.FrozenInstanceError`

(F401)